src/dispatch/incident/service.py

```diff
@@ -25,41 +24,53 @@
 SECONDS_IN_HOUR = 3600
 
 
-def resolve_incident_commander_email(
+def assign_incident_role(
     db_session: SessionLocal,
+    incident: Incident,
     reporter_email: str,
-    incident_type: str,
-    incident_name: str,
-    incident_title: str,
-    incident_description: str,
-    page_commander: bool,
+    role: ParticipantRoleType,
 ):
-    """Resolves the correct incident commander email based on given parameters."""
-    # We resolve the incident commander email
+    """Assigns incident roles."""
+    # We resolve the incident role email
     # default to reporter if we don't have an oncall plugin enabled
+    assignee_email = None
+
     oncall_plugin = plugin_service.get_active(db_session=db_session, plugin_type="oncall")
     if not oncall_plugin:
-        return reporter_email
+        assignee_email = reporter_email
 
-    incident_type = incident_type_service.get_by_name(db_session=db_session, name=incident_type)
+    if role == ParticipantRoleType.incident_commander:
+        # default to reporter
+        if not incident.incident_type.commander_service:
+            assignee_email = reporter_email
+        service = incident.incident_type.commander_service
+    else:
+        if not incident.incident_type.liaison_service:
+            return
+        service = incident.incident_type.liaison_service
 
-    if not incident_type.commander_service:
-        return reporter_email
+    if not assignee_email:
+        assignee_email = oncall_plugin.instance.get(service_id=service.external_id)
 
-    commander_service = incident_type.commander_service
+    # Add a new participant (duplicate participants with different roles will be updated)
+    participant_flows.add_participant(
+        assignee_email,
+        incident.id,
+        db_session,
+        role,
+    )
 
-    # page for high priority incidents
-    # we could do this at the end but it seems pretty important...
-    if page_commander:
+    # NOTE this will page both the commander and the liaison (if defined)
+    # we can either break these into two seperate flags or assume that if
+    # the commander gets paged the liaison should as well.
+    if incident.incident_priority.page_commander:
         oncall_plugin.instance.page(
-            service_id=commander_service.external_id,
-            incident_name=incident_name,
-            incident_title=incident_title,
-            incident_description=incident_description,
+            service_id=service.external_id,
+            incident_name=incident.name,
+            incident_title=incident.title,
+            incident_description=incident.description,
         )
```

Comment:
Should we move it inside the `if role == ParticipantRoleType.incident_commander` above and check if we have an `oncall_plugin` before paging?